src/main/java/docSharing/controller/UserController.java

```diff
@@ -17,26 +20,94 @@ public class UserController {
+
+        Optional<User> user = userService.getByEmail(email);
+        return user.map(value -> ResponseEntity.ok(BaseResponse.success(value)))
+                .orElseGet(() -> ResponseEntity.badRequest().body(BaseResponse.failure("User not found!")));
+    }
 
+    @RequestMapping(method = RequestMethod.DELETE, value = "delete")
+    public ResponseEntity<BaseResponse<String>> deleteUser(@RequestParam String email, @RequestHeader String token){
+        logger.debug("in deleteUser");
 
-    @RequestMapping(method = RequestMethod.POST)
-    public ResponseEntity<String> createUser(@RequestBody User user){
-        try {
-            return new ResponseEntity<>(userService.addUser(user).toString(), HttpStatus.OK);
-        } catch (SQLDataException e) {
-            throw new ResponseStatusException(
-                    HttpStatus.BAD_REQUEST, "Email already exists", e);
+        int id = authService.getUserIdByToken(email, token);
+        if (id == 0) {
+            return ResponseEntity.ok(BaseResponse.failure("User not authorized"));
         }
+
+        if (userService.deleteUser(id)) {
+            return ResponseEntity.ok(BaseResponse.noContent(true, "user " + email + " deleted"));
+        }
+
+        return ResponseEntity.ok(BaseResponse.failure("Delete user #" + id + "failed"));
     }
 
-    @RequestMapping(method = RequestMethod.GET)
+    @RequestMapping(method = RequestMethod.PATCH, value="/update/{email}", params = "name")
+    public ResponseEntity<BaseResponse<User>> updateName(@PathVariable("email") String email, @RequestHeader String token,
+                                                         @RequestParam String name) {
+        logger.debug("in updateName");
+
+        if (!Validate.isValidName(name)) {
+            return ResponseEntity.badRequest().body(BaseResponse.failure("Invalid name!"));
+        }
+
+        int id = authService.getUserIdByToken(email, token);
+        logger.debug(id);
+        if (id == 0) {
+            return ResponseEntity.ok(BaseResponse.failure("User not authorized"));
+        }
 
-    public ResponseEntity<User> getUserById(@RequestParam int id){
-        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        Optional<User> updatedUser = userService.updateName(id, name);
+        return updatedUser.map(value -> ResponseEntity.ok(BaseResponse.success(value))).
+                orElseGet(() -> ResponseEntity.ok(BaseResponse.failure("User not found")));
     }
 
-    @RequestMapping(value="/delete/{id}")
-    public ResponseEntity<?> deleteUserById(@PathVariable("id") int id){
-        return ResponseEntity.noContent().build();
+    @RequestMapping(method = RequestMethod.PATCH, value="/update/{email}", params = "newEmail")
+    public ResponseEntity<BaseResponse<User>> updateEmail(@PathVariable("email") String email, @RequestHeader String token,
+                                                          @RequestParam String newEmail){
+        logger.debug("in updateName");
```

Comment:
תשני ל:
 logger.debug("in updateEmail");

Reply:
טופל